### PR TITLE
[Fix] added compare taken leaves to available leaves, added validation

### PR DESCRIPTION
**Pull request by AntonMajest**
https://trello.com/c/6JUCJNsZ/235-extra-sick-leave-counter-not-working-also-vacation-and-sick-leave-counter-turn-negative-numbers-when-amount-exceeded
https://trello.com/c/Y7gGZMA1/119-profile-leaves-monday
image

### CODE REVIEW

File: web/src/components/admin/UserPage/UserLeavesTab.tsx [hunk 1]
```diff
@@ -59,18 +59,18 @@ const UserLeavesTab: React.FC<Props> = ({ leaves }) => {
 
   const leavesByTypes = () => {
     const categoryCount = {
-      sickLeave: availableSickLeave,
+      sickLeave: 0,
       extraSickLeave: 0,
-      vacation: availableVacation,
+      vacation: 0,
```
Comment (asmlvsk) [inline web/src/components/admin/UserPage/UserLeavesTab.tsx:64]: Числа винести

File: app/controllers/concerns/validate_vacations.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    def validate_vacation_on_create
+        record = params[:request]
+        records = Record.where(:user_id => record[:user_id]).where(:vacation => true).count
+    if record[:vacation]
+        if current_user.is_admin? && records.to_i >= 28
+        render json: { error: 'You have more than availables leave' }, status: :unprocessable_entity
+        end
+    end
+    end
+  end
```
Comment (asmlvsk) [inline app/controllers/concerns/validate_vacations.rb:12]: Пусту строку наче просить

File: app/controllers/concerns/validate_vacations.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+module ValidateVacations
+    def validate_vacation_on_create
+        record = params[:request]
+        records = Record.where(:user_id => record[:user_id]).where(:vacation => true).count
+    if record[:vacation]
+        if current_user.is_admin? && records.to_i >= 28
```
Comment (djmaluy) [inline app/controllers/concerns/validate_vacations.rb:7]: Число в константу винести

File: web/src/components/admin/UserPage/UserLeaveTabCard.tsx [hunk 4]
```diff
@@ -24,7 +25,11 @@ const UserLeavesTabCard: React.FC<Props> = ({ icon, counter, alt, title }) => {
         }}
       >
         <img src={icon} alt={alt} style={{ width: 50, height: 50, marginRight: '10px' }} />
-        <span>{counter}</span>
+        {availableLeave ? (
```
Comment (vtstepura) [inline web/src/components/admin/UserPage/UserLeaveTabCard.tsx:28]: <span>{counter} { availableLeave && `/ ${availableLeave}`}</span>

**Pull request merged**